bundles/org.openhab.binding.easee/src/main/java/org/openhab/binding/easee/internal/Utils.java

```diff
@@ -0,0 +1,197 @@
+    }
+
+    /**
+     * get element as String.
+     *
+     * @param jsonObject
+     * @param key
+     * @return
+     */
+    public static @Nullable String getAsString(@Nullable JsonObject jsonObject, String key) {
+        JsonElement element = jsonObject == null ? null : jsonObject.get(key);
+        String text = null;
+        if (element != null) {
+            if (element instanceof JsonPrimitive) {
+                text = element.getAsString();
+            } else if (element instanceof JsonObject || element instanceof JsonArray) {
+                text = element.toString();
+            }
+        }
+        return text;
+    }
+
+    /**
+     * get element as int.
+     *
+     * @param jsonObject
+     * @param key
+     * @return
+     */
+    public static int getAsInt(@Nullable JsonObject jsonObject, String key) {
+        JsonElement element = jsonObject == null ? null : jsonObject.get(key);
+        return (element instanceof JsonPrimitive) ? element.getAsInt() : 0;
+    }
+
+    /**
+     * get element as boolean.
+     *
+     * @param jsonObject
+     * @param key
+     * @return
+     */
+    public static @Nullable Boolean getAsBool(@Nullable JsonObject jsonObject, String key) {
+        JsonElement json = jsonObject == null ? null : jsonObject.get(key);
+        return (json == null || json instanceof JsonNull) ? null : json.getAsBoolean();
+    }
+
+    /**
+     * retrieves typeID of a channel.
+     *
+     * @param channel
+     * @return typeID or empty string if typeUID is null.
+     */
+    public static String getChannelTypeId(Channel channel) {
+        ChannelTypeUID typeUID = channel.getChannelTypeUID();
+        if (typeUID == null) {
+            return "";
+        }
+        return typeUID.getId();
+    }
+
+    /**
+     * retrieves the validation expression which is assigned to this channel, fallback to a public static, if no
+     * validation
+     * is
+     * defined.
+     *
+     * @param channel
+     * @return the validation expression
+     */
+    public static String getValidationExpression(Channel channel) {
+        String expr = getPropertyOrParameter(channel, PARAMETER_NAME_VALIDATION_REGEXP);
+        if (expr == null) {
+            // logger.warn("Channel {} does not have a validation expression configured", channel.getUID().getId());
+            throw new ConfigurationException(
+                    "channel (" + channel.getUID().getId() + ") does not have a validation expression configured");
+        }
+        return expr;
+    }
+
+    /**
+     * retrieves the write API url suffix which is assigned to this channel.
+     *
+     * @param channel
+     * @return the url suffix
+     */
+    public static String getWriteCommand(Channel channel) {
+        String command = getPropertyOrParameter(channel, PARAMETER_NAME_WRITE_COMMAND);
+        if (command == null) {
+            // logger.warn("channel {} does not have a write command configured", channel.getUID().getId());
```

Comment:
no. The binding is designed in a generic way and a configuration is necessary to determine which command has to be used for sending data. I thought that it is good idea to store this in the channel configuration rather than hard-coding this.

Reply:
If it is just related to the logging output, yes I will remove it.